lib/ops/opsservice/configure.go

```diff
@@ -115,6 +116,16 @@ func (o *Operator) ConfigurePackages(req ops.ConfigurePackagesRequest) error {
 		}
 		return trace.Wrap(err)
 	}
+
+	if operation.Type == ops.OperationExpand {
+		// TODO (knisbet) temporary timeout since configure packages overall isn't time limited
+		c, cancel := context.WithTimeout(context.TODO(), 5*time.Minute)
+		defer cancel()
+		err = o.registerKubernetesNode(c, *operation)
```

Comment:
Honestly, because invoking a phase on a master node (which has admin certs) is completely broken, and I gave up on figuring out why because nothing else within the join seems to use it. 

To me it makes sense to run the registration when creating the node within gravity, that it also creates it within kubernetes. This can be done during join, since kubernetes is already available. On install though, the node object needs to specifically be created after the kubernetes api starts, but before any pods are created.

Reply:
Hmm, can you define "completely broken"? :) I understand it's easier to just put it wherever but I'd really like us to use this phase-based approach, i.e. to me registering the node doesn't seem to belong to the "configure" phase.

I don't think it's actually even broken - join actually runs stuff on a master node via an agent when the 2nd master joins. Maybe in your case the agent wasn't just launched on the master? It is currently only launched when expanding 1->2 so chances are you just need to reorder the "startAgent" phase so it's executed before "registerNode". Unless it's some other error?

Reply:
This does use the phase based approach, it's just during the phase that creates the node within the site object, the node also gets created within kubernetes. It's happening in code that already has to run to join a node to the cluster, and is related to node creation within the cluster since this is where the node gets added to the site object.

The remote execution on join is indeed broken, I wasted a bunch of time checking this. When I looked through the join code, I didn't see any phases that actually are set to run on the master node. Instead, in the 1->2 scenario you describe, the phase is set to run on the joining node, but does an arbitrary command execution on the master node. Not the same thing as running a phase on the master. I didn't save the logs, but as I remember it, the phase execution is broken on remote nodes, which appears to be due to slight differences in the install/update agent behaviour. The join process is launching the agent as if it's an install, and as such has a problem actually starting the phase within the agent process (I didn't keep the logs).  Because nothing else actually uses this, I'm not sure it's worth the investment and risk of breaking something else.

The other thing to consider, is as I've mentioned before and flagged in the most recent security audit, we shouldn't be enabling arbitrary command executions between workers to masters at all. The only reason this got rated low, is because the agent is off most of the time and only started when in use. The remote phase invocation and the current behaviour you described both rely on arbitrary command executions, and probably should be re-considered.

Reply:
I guess if we treat this phase as "node creation", it sort of does make sense then. The only thing is should we then update the rollback for this phase to remove the Kubernetes node object too? Don't think I saw it.